### PR TITLE
Dev start agent dashboard

### PR DESCRIPTION
**Pull request by trevorhuis-smoothstack**
Ignore the code to load stuff for now, it is still rough from rushing to the demo. The login page is done and I created a skeleton for a dashboard.

### CODE REVIEW

File: src/app/agent/agent-dashboard/agent-dashboard.component.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  constructor(private http: HttpClient) { }
+
+  ngOnInit() {
+    this.loadFlights();
+  }
+
+  loadFlights() {
+    this.http.get("http://127.0.0.1:8083/agent/flights").subscribe(
+      (res) => {
+        console.log(res);
```
Comment (sean-darsie) [inline src/app/agent/agent-dashboard/agent-dashboard.component.ts:133]: should set some variable to the result.

File: src/common/h/service/AgentAuthService.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import * as moment from "moment";
+
+@Injectable()
+export class AgentAuthService {
+  constructor(private http: HttpClient) {}
+
+  login(username: string, password: string) {
+    return this.http
+      .post("http://127.0.0.1:8083/login", { username, password }, { observe: 'response' }).subscribe((response: any) =>{
```
Comment (sean-darsie) [inline src/common/h/service/AgentAuthService.ts]: is port 8083 supposed to be the orchestrator? i thought it was going to be 8080?

File: src/app/agent/agent-dashboard/agent-dashboard.component.html [hunk 3]
```diff
@@ -0,0 +1,89 @@
+<div id="agent-sidebar" class="active">
+  <ul *ngIf="user">
+    <li class="agent-data">Agent:<br>{{user.name}}</li>
+    <li class="agent-logout"><button type="button" class="btn btn-primary" (click)="logout()">Logout</button></li>
+    <li class="side-link" (click)="openBookFlight()">Book Flight</li>
+    <li class="side-link" (click)="openCancelFlight()">Cancel Flight</li>
```
Comment (justin-obrien-smoothstack) [inline src/app/agent/agent-dashboard/agent-dashboard.component.html:6]: "Cancel Booking" or "Cancel Ticket" would probably be a better choice of words than "Cancel Flight".

File: src/app/agent/agent-dashboard/agent-dashboard.component.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+      },
+      (error) => {
+        alert(error);
+      }
+    );
+  }
+
+  logout() {
+    this.authService.logout();
+    this.router.navigate(["/agent/login"]);
```
Comment (justin-obrien-smoothstack) [inline src/app/agent/agent-dashboard/agent-dashboard.component.ts:143]: router.navigate? I'm glad I know about that now.

File: src/app/agent/agent-login/agent-login.component.html [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+</div> -->
+
+<div class="agent-login">
+  <div class="login-form">
+    <form [formGroup]="form">
+      <fieldset>
+        <legend>Agent Login</legend>
+        <div class="form-group">
+          <label>Username</label>
+          <input name="email" class="form-control" formControlName="username" />
```
Comment (justin-obrien-smoothstack) [inline src/app/agent/agent-login/agent-login.component.html:16]: name="email" could be a bit misleading here.

File: src/app/common/h/service/AgentAuthService.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  logout() {
+    localStorage.removeItem("id_token_agent");
+    localStorage.removeItem("expires_at");
+    localStorage.removeItem("username");
+  }
+
+  public isLoggedIn() {
+    return moment().isBefore(this.getExpiration().add(1, "hours"));
```
Comment (justin-obrien-smoothstack) [inline src/app/common/h/service/AgentAuthService.ts:26]: Why add an hour?

Comment (trevorhuis-smoothstack) [reply src/app/common/h/service/AgentAuthService.ts:26]: The same time stuff was happening from the database. By the time it hit the front end, it was already expired. I do not think this is a good final solution but it worked.

File: src/app/agent/agent-dashboard/agent-dashboard.component.html (hunk 3 above)
Comment (trevorhuis-smoothstack) [reply src/app/agent/agent-dashboard/agent-dashboard.component.html:6]: good call

**Comment by trevorhuis-smoothstack** [comment]
I think the images are supposed to be stored in an s3 bucket but I didn't want to pay for/set that up till later. I think it is fine to be in the repo for now since it's not sensitive. I plan on making all alerts toasts, not modals since that requires another click from a user.

**Pull request merged**